config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml

```diff
@@ -1264,55 +1264,49 @@ presubmits:
             cpu: 4
             memory: 6Gi
 
-  # This jobs runs e2e.test with a focus on tests for the Dynamic Resource Allocation feature (currently alpha).
-  # crio >= 1.23.0 has the necessary support for Container Device Interface (CDI).
-  - name: pull-kubernetes-node-crio-dra
-    cluster: k8s-infra-prow-build
+  # This jobs runs e2e.test with a focus on tests for the Dynamic Resource Allocation feature (currently alpha)
+  # on a kind cluster with containerd updated to a version with CDI support.
+  - name: pull-kubernetes-kind-dra
     skip_branches:
     - release-\d+\.\d+  # per-release image
     annotations:
       testgrid-dashboards: sig-node-presubmits
       testgrid-tab-name: pr-crio-dra
+    decorate: true
     # Not relevant for most PRs.
     always_run: false
     # This covers most of the code related to dynamic resource allocation.
-    run_if_changed: /dra/|/dynamicresources/|/resourceclaim/|/resourceclass/|/podscheduling/
+    run_if_changed: /dra/|/dynamicresources/|/resourceclaim/|/resourceclass/|/podscheduling/|/dynamic-resource-allocation/
     # The tests might still be flaky or this job might get triggered accidentally for
     # an unrelated PR.
     optional: true
-    max_concurrency: 12
     labels:
       preset-service-account: "true"
-      preset-k8s-ssh: "true"
-      preset-pull-kubernetes-e2e: "true"
-      preset-pull-kubernetes-e2e-gce: "true"
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
     spec:
       containers:
       - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220928-cd48f52a16-master
+        command:
+        - runner.sh
         args:
-        - --root=/go/src
-        - "--job=$(JOB_NAME)"
-        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
-        - "--service-account=/etc/service-account/service-account.json"
-        - "--upload=gs://kubernetes-jenkins/pr-logs"
-        - "--timeout=240"
-        - --scenario=kubernetes_e2e
-        - -- # end bootstrap args, scenario args below
-        - --deployment=node
-        - --env=KUBE_SSH_USER=core
-        - --env=KUBE_FEATURE_GATES=DynamicResourceAllocation=true
-        - --gcp-zone=us-west1-b
-        - '--test_args=--nodes=4 --focus="\[Feature:DynamicResourceAllocation\]"'
-        - --provider=gce
-        - --timeout=180m
-        - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/crio/latest/image-config-cgrpv1-serial.yaml
+        - /bin/sh
+        - -c
+        - >
+          make WHAT=github.com/onsi/ginkgo/v2/ginkgo &&
+          make WHAT=k8s.io/kubernetes/test/e2e/e2e.test &&
+          go install sigs.k8s.io/kind@main &&
```

Comment:
FWIW we have pre-built binaries. Bit faster / less log noise, but more verbose in the job.

One approach:
https://github.com/kubernetes/test-infra/blob/70b0eedb87ef5f61e2559ef95604c9b7ecb55e01/config/jobs/kubernetes-sigs/kind/kind.yaml#L50

Reply:
Does that pull the latest stable release or a build of "main"?

Because we need to build Kubernetes master, we cannot use the latest stable release - it might not work for Kubernetes master anymore. We do the same for Kubernetes-CSI jobs.

We have to live with occasional breakage because of new kind bugs, but those should get fixed quickly (not that those ever occur, right?).

Reply:
latest is perhaps a poor name since go module tags. It is built on postsubmit so ~= HEAD

Reply:
This is what we use to test k/k for the reasons you mentioned above. We try very hard to at least not have breaking changes for that type of workflow (where node images are being built and run)

Reply:
Makes sense. I've changed to downloading the pre-built kind.

Reply:
I also realized that the output probably wouldn't have been very readable without dumping commands, so I added "-x" to the `/bin/sh` options.